### PR TITLE
Decouple frontend UI root subproject from frontend UI table implementation

### PR DESCRIPTION
**Pull request by PawelLipski**
Another step necessary for #220

### CODE REVIEW

File: binding/config/checkstyle/checkstyle.xml [hunk 1]
```diff
@@ -0,0 +1 @@
+../../../config/checkstyle/checkstyle_allow_java_collections.xml
```
Comment (PawelLipski) [inline binding/config/checkstyle/checkstyle.xml:1]: Symlinks everywhere

File: config/checkstyle/checkstyle.xml [hunk 2]
```diff
@@ -23,6 +23,7 @@
     </module>
 
     <module name="TreeWalker">
+        &prohibitJavaCollections;
```
Comment (PawelLipski) [inline config/checkstyle/checkstyle.xml:26]: Now I assume that Java collections are prohibited by default, you need to opt-out from the prohibition (previously it was opt-in)

File: frontendUiApi/build.gradle [hunk 3]
```diff
@@ -0,0 +1,8 @@
+vavr()
```
Comment (PawelLipski) [inline frontendUiApi/build.gradle:1]: This module is basically for the stuff that `frontendUiTableImpl` and `frontendUiRootImpl` need to pass b/w each other (dropdown, table manager)

File: frontendUiRootImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/root/VcsRootComboBox.java [hunk 4]
```diff
@@ -12,23 +11,24 @@
...
-  private final List<ISelectionChangeObserver> observers = new SmartList<>();
+public final class VcsRootComboBox extends JComboBox<GitRepository> implements IGitRepositorySelectionProvider {
+
+  private final java.util.List<IGitRepositorySelectionChangeObserver> observers = new SmartList<>();
 
   /**
    * @param repositories non-empty list of {@link git4idea.repo.GitRepository} that represents VCS repositories
    */
   @UIEffect
   public VcsRootComboBox(@MinLen(1) List<GitRepository> repositories) {
```
Comment (PawelLipski) [inline frontendUiRootImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/root/VcsRootComboBox.java:29]: `MinLen` didn't make much sense on a mutable collection :/

File: frontendUiRootImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/root/GitMachetePanel.java [hunk 5]
```diff
@@ -36,28 +35,28 @@
...
   private static final String TOGGLE_LIST_COMMIT_DESCRIPTION = "Toggle list commits";
 
-  private final GitMacheteGraphTableManager gitMacheteGraphTableManager;
+  private final IGraphTableManager gitMacheteGraphTableManager;
 
   @UIEffect
   public GitMachetePanel(Project project) {
     super(/* vertical */ false, /* borderless */ true);
-    LOG.debug("Instantiation of GitMachetePanel");
+    LOG.debug("Instantiation of ${getClass().getSimpleName()}");
```
Comment (micpiotrowski) [inline frontendUiRootImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/root/GitMachetePanel.java:43]: Wrt. #214 that has been merged something like `LOG.debug("Instantiation")` is enough

Comment (PawelLipski) [reply frontendUiRootImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/root/GitMachetePanel.java:43]: Fixed on PR #233

**Pull request merged**